sys/crypto/ciphers.c

```diff
@@ -22,30 +22,24 @@
 int cipher_init(cipher_t *cipher, cipher_id_t cipher_id, const uint8_t *key,
                 uint8_t key_size)
 {
-    if (key_size > cipher_id->max_key_size) {
-        return CIPHER_ERR_INVALID_KEY_SIZE;
-    }
-
     cipher->interface = cipher_id;
-    return cipher->interface->init(&cipher->context, key, key_size);
+    cipher->context.key_size = key_size;
```

Comment:
I would rather deal with that in a follow-up PR and focus on `aes_*` here (see comment about API change above)